### PR TITLE
Fix eigen decomposition; new feature to store all optimization tries; and updates to documentation

### PR DESCRIPTION
**Pull request by d-schindler**
I made a few changes to the code:

- Fix: Replace linalg.eig with linalg.eigh for the spectral decomposition used for computation of exponentials. We only apply spectral decompostion in the case of undirected graphs where linalg.eigh is the faster method (and we actually found an example where linalg.eig does produce incorrect results for an undirected graph).
- Feature: I introduced a parameter `with_all_tries` in the pgs.run() function that stores the information from all optimization tries in the results dictionary under an additional key "all_tries". For some applications it is interesting to look at the different optimizations produced for the same scale, not only at the best one that is selected.
- Fix: There are a few things I corrected/updated in the documentation.

### CODE REVIEW

**Comment by d-schindler** [comment]
I also renamed `block_detection_curve` in the optimal scale selection to `block_nvi` to make it consistent with our figure labels and the terminology in our software paper.

**Comment by d-schindler** [comment]
I added a figsize parameter to the `plot_scan` function, which is something I found quite useful.

**Comment by d-schindler** [comment]
@arnaudon , before we merge this PR we should probably update the online documentation of the module.

**Comment by d-schindler** [comment]
Further improvement of the documentation: I added the missing division by <d> in the formula for MS with the continuous combinatorial Laplacian.

**Comment by d-schindler** [comment]
I've also added more references to the formulas for constructors. @arnaudon , I used reference [1] several times and wasn't sure if I should repeat it in every docstring or not.

File: src/pygenstability/plotting.py [hunk 1]
```diff
@@ -420,9 +447,12 @@ def _plot_optimal_scales(all_results, ax, scales, ax1, ax2):
         ax2.axvline(scale, ls="--", color="C4")
 
 
-def plot_scan_plt(all_results, scale_axis=True, figure_name="scan_results.svg"):
+def plot_scan_plt(
+    all_results, figsize=(6, 5), scale_axis=True, figure_name="scan_results.svg"
+):
     """Plot results of pygenstability with matplotlib."""
     scales = _get_scales(all_results, scale_axis=scale_axis)
+    plt.figure(figsize=figsize)
```
Comment (arnaudon) [inline src/pygenstability/plotting.py:455]: I think I didn't put this here, so one can call figure before this function, and have more freedom

**Pull request merged**